### ISSUE TITLE
mert_path

### ISSUE THREAD

**Issue by RevolGMPHL** [crypto-code/MU-LLaMA#7]
i got this error: LLaMA_adapter.__init__() missing 1 required positional argument: 'mert_path'
which one is the  mert_path?

**Comment by shansongliu** [comment]
Can you provide a more detailed error report?

**Comment by GMPHL** [comment]
i see the error in code:    model = LLaMA_adapter(
        llama_ckpt_dir, llama_tokenzier_path, knn=knn, phase=phase)
but the class LLaMA_adapter have 3 path  :__init__(self, llama_ckpt_dir, llama_tokenizer, mert_path,

**Comment by RevolGMPHL** [comment]
i see the error in this code: 

model = LLaMA_adapter(llama_ckpt_dir, llama_tokenzier_path, knn=knn, phase=phase)

but the class LLaMA_adapter have 3 path :init(self, llama_ckpt_dir, llama_tokenizer, mert_path,

**Comment by shansongliu** [comment]
> i see the error in this code:
> 
> model = LLaMA_adapter(llama_ckpt_dir, llama_tokenzier_path, knn=knn, phase=phase)
> 
> but the class LLaMA_adapter have 3 path :init(self, llama_ckpt_dir, llama_tokenizer, mert_path,

Can you copy and paste the full traceback error report by the Python interpreter?

**Comment by RevolGMPHL** [comment]
> > i see the error in this code:
> > model = LLaMA_adapter(llama_ckpt_dir, llama_tokenzier_path, knn=knn, phase=phase)
> > but the class LLaMA_adapter have 3 path :init(self, llama_ckpt_dir, llama_tokenizer, mert_path,
> 
> Can you copy and paste the full traceback error report by the Python interpreter?



Traceback (most recent call last):
  File "/disk_1/aaa/MU-LLaMA/MU-LLaMA/gradio_app.py", line 29, in <module>
    model = llama.load(args.model, args.llama_dir, knn=True, llama_type=args.llama_type)
  File "/disk_1/aaa/MU-LLaMA/MU-LLaMA/llama/llama_adapter.py", line 398, in load
    model = LLaMA_adapter(
TypeError: LLaMA_adapter.__init__() missing 1 required positional argument: 'mert_path'

**Comment by shansongliu** [comment]
> > > i see the error in this code:
> > > model = LLaMA_adapter(llama_ckpt_dir, llama_tokenzier_path, knn=knn, phase=phase)
> > > but the class LLaMA_adapter have 3 path :init(self, llama_ckpt_dir, llama_tokenizer, mert_path,
> > 
> > 
> > Can you copy and paste the full traceback error report by the Python interpreter?
> 
> Traceback (most recent call last): File "/disk_1/aaa/MU-LLaMA/MU-LLaMA/gradio_app.py", line 29, in model = llama.load(args.model, args.llama_dir, knn=True, llama_type=args.llama_type) File "/disk_1/aaa/MU-LLaMA/MU-LLaMA/llama/llama_adapter.py", line 398, in load model = LLaMA_adapter( TypeError: LLaMA_adapter.**init**() missing 1 required positional argument: 'mert_path'

I will ask the owner to fix this issue, it seems that the gradio_app.py has not been updated.

**Comment by RevolGMPHL** [comment]
> > > > i see the error in this code:
> > > > model = LLaMA_adapter(llama_ckpt_dir, llama_tokenzier_path, knn=knn, phase=phase)
> > > > but the class LLaMA_adapter have 3 path :init(self, llama_ckpt_dir, llama_tokenizer, mert_path,
> > > 
> > > 
> > > Can you copy and paste the full traceback error report by the Python interpreter?
> > 
> > 
> > Traceback (most recent call last): File "/disk_1/aaa/MU-LLaMA/MU-LLaMA/gradio_app.py", line 29, in model = llama.load(args.model, args.llama_dir, knn=True, llama_type=args.llama_type) File "/disk_1/aaa/MU-LLaMA/MU-LLaMA/llama/llama_adapter.py", line 398, in load model = LLaMA_adapter( TypeError: LLaMA_adapter.**init**() missing 1 required positional argument: 'mert_path'
> 
> I will ask the owner to fix this issue, it seems that the gradio_app.py has not been updated.

thx~~~

**Comment by crypto-code** [comment]
I have fixed the issue (9fd1dac727cc8e7e2ed00eea8cd57354a7388ba9) by adding the MERT argument to the ```load``` function in the```llama_adapter.py``` file.

```python
# The model files for MERT can be downloaded here in case of network issues:
# https://huggingface.co/m-a-p/MERT-v1-330M
# And set the MERT argument to directory with the model files
model = LLaMA_adapter(llama_ckpt_dir, llama_tokenzier_path, "m-a-p/MERT-v1-330M", knn=knn, phase=phase)
```

I hope this will fixed the issue for you.

**Comment by shansongliu** [comment]
> I have fixed the issue ([9fd1dac](https://github.com/crypto-code/MU-LLaMA/commit/9fd1dac727cc8e7e2ed00eea8cd57354a7388ba9)) by adding the MERT argument to the `load` function in the`llama_adapter.py` file.
> 
> ```python
> # The model files for MERT can be downloaded here in case of network issues:
> # https://huggingface.co/m-a-p/MERT-v1-330M
> # And set the MERT argument to directory with the model files
> model = LLaMA_adapter(llama_ckpt_dir, llama_tokenzier_path, "m-a-p/MERT-v1-330M", knn=knn, phase=phase)
> ```
> 
> I hope this will fixed the issue for you.

Great, do we still allow customizable input path for the pretrained model like MERT? Since some users may tend to use offline downloaded model.

**Comment by crypto-code** [comment]
> > I have fixed the issue ([9fd1dac](https://github.com/crypto-code/MU-LLaMA/commit/9fd1dac727cc8e7e2ed00eea8cd57354a7388ba9)) by adding the MERT argument to the `load` function in the`llama_adapter.py` file.
> > ```python
> > # The model files for MERT can be downloaded here in case of network issues:
> > # https://huggingface.co/m-a-p/MERT-v1-330M
> > # And set the MERT argument to directory with the model files
> > model = LLaMA_adapter(llama_ckpt_dir, llama_tokenzier_path, "m-a-p/MERT-v1-330M", knn=knn, phase=phase)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I hope this will fixed the issue for you.
> 
> Great, do we still allow customizable input path for the pretrained model like MERT? Since some users may tend to use offline downloaded model.

I have added command-line arguments for the path to the MERT model (ebc5c78032258710e288ff3d7b53acede8cdf2fd). This can be set to the path with the downloaded MERT model.

**Comment by shansongliu** [comment]
> > > I have fixed the issue ([9fd1dac](https://github.com/crypto-code/MU-LLaMA/commit/9fd1dac727cc8e7e2ed00eea8cd57354a7388ba9)) by adding the MERT argument to the `load` function in the`llama_adapter.py` file.
> > > ```python
> > > # The model files for MERT can be downloaded here in case of network issues:
> > > # https://huggingface.co/m-a-p/MERT-v1-330M
> > > # And set the MERT argument to directory with the model files
> > > model = LLaMA_adapter(llama_ckpt_dir, llama_tokenzier_path, "m-a-p/MERT-v1-330M", knn=knn, phase=phase)
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > I hope this will fixed the issue for you.
> > 
> > 
> > Great, do we still allow customizable input path for the pretrained model like MERT? Since some users may tend to use offline downloaded model.
> 
> I have added command-line arguments for the path to the MERT model ([ebc5c78](https://github.com/crypto-code/MU-LLaMA/commit/ebc5c78032258710e288ff3d7b53acede8cdf2fd)). This can be set to the path with the downloaded MERT model.

Good! See if this fixed your encountered problem @GMPHL . You can close the issue if this is resolved. We would appreciate you starring our repo 😊.

**Comment by RevolGMPHL** [comment]
> > > > I have fixed the issue ([9fd1dac](https://github.com/crypto-code/MU-LLaMA/commit/9fd1dac727cc8e7e2ed00eea8cd57354a7388ba9)) by adding the MERT argument to the `load` function in the`llama_adapter.py` file.
> > > > ```python
> > > > # The model files for MERT can be downloaded here in case of network issues:
> > > > # https://huggingface.co/m-a-p/MERT-v1-330M
> > > > # And set the MERT argument to directory with the model files
> > > > model = LLaMA_adapter(llama_ckpt_dir, llama_tokenzier_path, "m-a-p/MERT-v1-330M", knn=knn, phase=phase)
> > > > ```
> > > > 
> > > > 
> > > >     
> > > >       
> > > >     
> > > > 
> > > >       
> > > >     
> > > > 
> > > >     
> > > >   
> > > > I hope this will fixed the issue for you.
> > > 
> > > 
> > > Great, do we still allow customizable input path for the pretrained model like MERT? Since some users may tend to use offline downloaded model.
> > 
> > 
> > I have added command-line arguments for the path to the MERT model ([ebc5c78](https://github.com/crypto-code/MU-LLaMA/commit/ebc5c78032258710e288ff3d7b53acede8cdf2fd)). This can be set to the path with the downloaded MERT model.
> 
> Good! See if this fixed your encountered problem @GMPHL . You can close the issue if this is resolved. We would appreciate you starring our repo 😊.

thx~~~it works
